plugin/keyring_vault/vault_credentials.cc

```diff
@@ -0,0 +1,18 @@
+#include "vault_credentials.h"
+
+namespace keyring
+{
+
+const Secure_string& get_credential(const Vault_credentials &credentials, const Secure_string &key)
+{
+  static Secure_string empty_value;
```

Comment:
1. Potential user of this function does not know that it is safe only because it's called once from a single-threaded location. There is no guarantee that somebody will be calling this function from other place in a few years.
2. It's not a problem with double initialization but It's a problem when it gets destroyed twice at shutdown.
3. You can avoid polluting the namespace by simply defining this `empty_value` as a static inside the `.cc` file (do not expose it in the `.h`). Alternatively, put it into an anonymous namespace inside the `.cc` file.